src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -209,4 +213,41 @@ public static Price parsePrice(String price) throws ParseException {
             throw new ParseException(MESSAGE_NO_ARGS);
         }
     }
+
+    /**
+     * Parses the keywords for each of the predicates. If parsing fails, the keyword is dropped as it isn't a valid
+     * search term.
+     * @param keywords search terms to be parsed.
+     * @param prefix prefix to operate on.
+     * @return list of valid keywords to search on for corresponding prefix.
+     */
+    public static List<String> parseAttributePredicateKeywords(List<String> keywords, String prefix) {
+        List<String> outputList = new ArrayList<>();
+
+        for (String s : keywords) {
+            try {
+                switch (prefix) {
+                case PREFIX_NAME_STRING:
+                    outputList.add(ParserUtil.parseName(s).toString());
+                    break;
+                case PREFIX_ADDRESS_STRING:
+                    outputList.add(ParserUtil.parseAddress(s).toString());
+                    break;
+                case PREFIX_PHONE_STRING:
+                    outputList.add(ParserUtil.parsePhone(s).toString());
+                    break;
+                case PREFIX_TAG_STRING:
+                    outputList.add(ParserUtil.parseTag(s).toString());
+                    break;
+                default:
+                    break;
+                }
+
+            } catch (ParseException ignored) {
```

Comment:
I don't know if you can do it like that? I might be wrong but isn't leaving an empty catch block not good practice?

Reply:
Ok my mistake looks like parseKeyword from the original AB3 does something similar.

Reply:
maybe just insert a continue?